source/slang/slang-ir-constexpr.cpp

```diff
@@ -195,6 +195,14 @@ void maybeAddToWorkList(
     {
         context->workList.add(gv);
         context->onWorkList.add(gv);
+
+        // Iterating uses to include generic functions
```

Comment:
Can you explain with an example what issue we are having and why this solves the problem?

Reply:
I can leave a more descriptive comment as requested.
But before that, I like to check it with you if I am doing the right thing.
The issue I am trying to address is described in [my comment](https://github.com/shader-slang/slang/issues/3834#issuecomment-2076069811).

Basically, it `module->getGlobalInsts()` doesn't include generic functions.
Only way I can iterate the generic functions is to iterate all of uses like how it is done in `SpecializationContext::addToWorkList()`.
Is this a right way to iterate generic functions?

Reply:
getGlobalInsts() should return all insts, including IRGeneric. However, the function body of a generic function is an IRFunc in the body of the generic.

You need to do:

if (auto gen=as<IRGeneric>(globalInst))
   auto innerFunc = as<IRFunc>(findGenericReturnVal(gen));

Reply:
Oh. I see.
I will try that.